internal/core/services/notification.go

```diff
@@ -0,0 +1,107 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/google/uuid"
+	core "github.com/iden3/go-iden3-core"
+	"github.com/iden3/go-schema-processor/verifiable"
+
+	"github.com/polygonid/sh-id-platform/internal/core/domain"
+	"github.com/polygonid/sh-id-platform/internal/core/ports"
+	"github.com/polygonid/sh-id-platform/internal/log"
+	"github.com/polygonid/sh-id-platform/pkg/notifications"
+	"github.com/polygonid/sh-id-platform/pkg/pubsub"
+)
+
+type notification struct {
+	issuerDID           core.DID
+	notificationGateway ports.NotificationGateway
+	connService         ports.ConnectionsService
+	credService         ports.ClaimsService
+}
+
+// NewNotification returns a Notification Service
+func NewNotification(issuerDID core.DID, notificationGateway ports.NotificationGateway, connService ports.ConnectionsService, credService ports.ClaimsService) ports.NotificationService {
+	return &notification{
+		notificationGateway: notificationGateway,
+		connService:         connService,
+		credService:         credService,
+		issuerDID:           issuerDID,
+	}
+}
+
+func (n *notification) SendCreateCredentialNotification(ctx context.Context, e pubsub.Event) error {
+	var cEvent pubsub.CreateCredentialEvent
+	if err := cEvent.FromEvent(e); err != nil {
+		return errors.New("sendCredentialNotification unexpected data type")
+	}
+
+	return n.sendCreateCredentialNotification(ctx, cEvent.ID)
+}
+
+func (n *notification) sendCreateCredentialNotification(ctx context.Context, credID string) error {
+	credUUID, err := uuid.Parse(credID)
+	if err != nil {
+		return err
```

Comment:
Log the credentiaID, etc... Any info to understand what is happening.

Reply:
sure, thanks!